core/features/core.js

```diff
@@ -0,0 +1,264 @@
+      Authorization: `Custom ${CUSTOM_TOKEN_ID}`,
+      'Cache-Control': 'no-cache'
+    }
+  }
+
+  const checkPatientExistsResponse = await axios(checkPatientExistsOptions)
+
+  if (checkPatientExistsResponse.data.total === 0) {
+    console.log(
+      `Patient record for Jane Doe does not exist. Creating Patient...`
+    )
+    const options = {
+      url: `${OPENHIM_PROTOCOL}://${OPENHIM_API_HOSTNAME}:${OPENHIM_TRANSACTION_API_PORT}/hapi-fhir-jpaserver/fhir/Patient`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Custom ${CUSTOM_TOKEN_ID}`,
+        'Cache-Control': 'no-cache'
+      },
+      data: {
+        resourceType: 'Patient',
+        name: {
+          use: 'temp',
+          family: 'Doe',
+          given: ['Jane']
+        },
+        identifier: {
+          use: 'temp',
+          value: 'test'
+        },
+        gender: 'male'
+      }
+    }
+
+    const createPatientResponse = await axios(options)
+
+    expect(createPatientResponse.status).to.eql(201)
+
+    hapiFhirPatientID = createPatientResponse.data.id
+  } else if (checkPatientExistsResponse.data.total === 1) {
+    console.log(`Patient record for Jane Doe already exists...`)
+
+    hapiFhirPatientID = checkPatientExistsResponse.data.entry[0].resource.id
+  } else {
+    // Previous test data should have been cleaned out
+    throw new Error(
+      `Multiple Patient records for Jane Doe exist: ${checkPatientExistsResponse.data.total}`
+    )
+  }
+})
+
+// Ensure OpenHIM Test Client exists
+BeforeAll(async function () {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0
+  const checkClientExistsOptions = {
+    url: `https://${OPENHIM_API_HOSTNAME}:${OPENHIM_MEDIATOR_API_PORT}/clients`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: BASIC_AUTH_HEADER
+    }
+  }
+
+  const checkClientExistsResponse = await axios(checkClientExistsOptions)
+
+  let createClient = true
+  // Previous test data should have been cleaned out
+  for (let client of checkClientExistsResponse.data) {
+    if (client.clientID === 'test-harness-client') {
+      createClient = false
+      break
+    }
+  }
+
+  if (createClient) {
+    console.log(`The test Harness Client does not exist. Creating Client...`)
+    const options = {
+      url: `https://${OPENHIM_API_HOSTNAME}:${OPENHIM_MEDIATOR_API_PORT}/clients`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: BASIC_AUTH_HEADER
+      },
+      data: {
+        roles: ['instant'],
+        clientID: 'test-harness-client',
+        name: 'Alice',
+        customTokenID: 'test-harness-token'
+      }
+    }
+
+    const response = await axios(options)
+    expect(response.status).to.eql(201)
+  } else {
+    console.log(`The Test Harness Client (Alice) already exists...`)
+  }
+})
+
+Given('a patient, Jane Doe, exists in the FHIR server', async function () {
```

Comment:
@MattyJ007 the intention of the given statements is to give you an opportunity to create the needed test data, so I think you can move the test data creation fro the before all to the given section. This should reduce the amount of checking that you need to do as well.

Reply:
> When Cucumber executes a Given step, it will configure the system to be in a well-defined state, such as creating and configuring objects or adding data to a test database.

https://cucumber.io/docs/gherkin/reference/

Reply:
Did this to avoid duplicate creation before I knew it was HAPI causing it :| 

Ge-sorted :+1: